soundbay/active_learning.py

```diff
@@ -0,0 +1,165 @@
+import argparse
+import numpy as np
+import pandas as pd
+import os
+
+
+def get_recording_name_from_inference_file_name(inference_file_name):
+    recording_name = inference_file_name.split('\\')[-1].split('.')[0].split('-')[-1]
+    return recording_name
+
+
+def load_inference_results_from_dir(inference_dir: str) -> pd.DataFrame:
+    df_all_recordings_inference = pd.DataFrame(
+        columns=['segment_id', 'recording', 'class0_prob', 'class1_prob', 'segment_start_sec', 'segment_end_sec'])
+
+    for filename in os.listdir(inference_dir):
+        df_one_recording_inference = create_inference_df_for_one_recording(filename, inference_dir)
+        df_all_recordings_inference = pd.concat([df_all_recordings_inference, df_one_recording_inference],
+                                                ignore_index=True)
+
+    df_all_recordings_inference['segment_end_sec'] = df_all_recordings_inference['segment_start_sec'] + 1
+    df_all_recordings_inference.insert(0, 'chunk_id', '')
+    return df_all_recordings_inference
+
+
+def create_inference_df_for_one_recording(filename, inference_dir):
+    """
+    Load inference file and create DataFrame for one recording.
+    :param filename: str
+    :param inference_dir: str
+    :return: pd.DataFrame
+    """
+    recording_name = get_recording_name_from_inference_file_name(filename)
+    inference_full_path = os.path.join(inference_dir, filename)
+    df_one_recording_inference = pd.read_csv(inference_full_path)
+    df_one_recording_inference.insert(0, 'recording', recording_name)
+    df_one_recording_inference['segment_start_sec'] = df_one_recording_inference.index
+    df_one_recording_inference['segment_id'] = df_one_recording_inference['recording'] + '_' + \
+                                               df_one_recording_inference[
+                                                   'segment_start_sec'].astype(
+                                                   str)
+    return df_one_recording_inference
+
+
+def validate_chunk_end_sec(chunk_end_sec: int, chunk_start_sec: int, chunk_actual_size: int) -> int:
+    """
+    Make sure the chunk_end_sec is correct (that it fits the chunk_start_sec and the desired chunk length).
+    Else, correct it.
+    """
+    if chunk_end_sec - chunk_start_sec == chunk_actual_size:
+        return chunk_end_sec
+    elif chunk_end_sec - chunk_start_sec > chunk_actual_size:
+        chunk_end_sec = chunk_start_sec + chunk_actual_size
+        return chunk_end_sec
+    else:
+        print('chunk size error!')
+        exit()
+
+
+def add_chunk_id_per_recording(df: pd.DataFrame, recording_id: str, chunk_len_in_seconds: int = 300) -> pd.DataFrame:
```

Comment:
I'm not sure I understood why this whole thing existis. 
My best guess currently is that "segment" are the lengths the file was cut into for inference, while chunks are a different segmentation of the file with a different time interval used to mark "checks" that are then prioritized for active learning.

Question 1 in that case - why not prioritize the original segments? 
Thought 2 - seems like there should be an easier way to do that (if this is indeed what is done...), I mean if both lenghs are fixed theres a simple mathematical conversion. I'm probably missunderstanding something though.